### DISCUSSION TITLE
Change RAILS_MAX_THREADS  (timeout api)

### DISCUSSION THREAD

**Discussion by adamito** [postalserver/postal#2479] [Help with using Postal]
Deal postalserver community

I would like to raise RAILS_MAX_THREADS for postal-web / puma because I'm getting many time timeouts.

My config now. : postal-web-1  | Puma starting in single mode...
postal-web-1  | * Puma version: 6.1.1 (ruby 3.2.1-p31) ("The Way Up")
postal-web-1  | *  Min threads: 5
postal-web-1  | *  Max threads: 5
postal-web-1  | *  Environment: production

Can you please give me any suggestion.

Best

**Comment by willpower232** [comment]
where are you getting timeouts specifically?

**Comment by Bob-emblibrary** [comment]
If you edit the postal.yml under /opt/postal/config and add the line 'max_threads: 10' under the webserver config, you'll get 10 threads.

**Comment by drraccoony** [comment]
@willpower232 We're experiencing timeouts as well, specifically when POST'ing to `/api/v1/send/message`.

We have a newsletter application that is building newsletters and sending jobs to Postal. Its fine when we have like 10 agents, but the second we go any higher than like 16, we start getting cURL timeouts to `/api/v1/send/message`.

It'd be awesome to figure out what we need to adjust to overcome these timeouts as its creating a rather large bottleneck for our workflow. :(

All help is greatly appreciated!

**Comment by drraccoony** [comment]
> If you edit the postal.yml under /opt/postal/config and add the line 'max_threads: 10' under the webserver config, you'll get 10 threads.

To add to this, We did this and that resolved the problem with API timing out, but keep in mind if you are trying to heavily increase the load of API calls for sending mailers, you may need to increase the connection pools.

Unrelated to the change, but since it could handle more API calls, we started getting the following error after:

> lib/postal/tracking_middleware.rb:34:in \`dispatch_image_request'lib/postal/tracking_middleware.rb:21:in \`call'104.28.58.4 - - [18/Aug/2023:16:23:06 +0000] "GET /500 HTTP/1.1" 500 1477 5.0037[10] [2023-08-18T16:23:06.843] FATAL -- :   ActiveRecord::ConnectionTimeoutError (could not obtain a connection from the pool within 5.000 seconds (waited 5.000 seconds); all pooled connections were in use)

We fixed this by editing `postal.yml` and making two changes.
1. under main_db: add `pool_size: 10`
2. under message_db: add the same, `pool_size: 10`

Citation: https://github.com/postalserver/postal/pull/751#issuecomment-466540422 & https://github.com/postalserver/postal/issues/324#issuecomment-323538575
